README.md

```diff
@@ -1,2 +1,221 @@
+use the default queue defined for that pipeline.
+
+```
+process {
+  withName: my_task_name {
+      queue = 'TowerForge-queue-id'
+
+      maxErrors     = '-1'
+      maxRetries    = 3
+      errorStrategy = { task.attempt <= 3 ? 'retry' : 'finish' }
+
+      cpus   = 2
+      memory = 1.GB
+  }
+}
+```
+
+The `queue` in this case comes from the Tower UI. First navigate to the workspace in
+question and go to the `Compute Environments` tab. Select the environment in question.
+
+![find-queue-id-1](./find-queue-id-1.png)
+
+Scroll down until you find the ID for `Compute queue`. This is that ID you'll use.
+
+![find-queue-id-2](./find-queue-id-2.png)
+
+<br>
+
+##### Additional example:
+
+The `maxSubmitAwait` directives allows you to specify how long a task can remain in submission queue without being executed. Elapsed this time the task execution will fail.
+
+When used along with `retry` error strategy, it can be useful to re-schedule the task to a difference queue or resource requirement. For example:
+
+```
+process foo {
+  errorStrategy 'retry'
+  maxSubmitAwait '10 mins'
+  maxRetries 3
+  queue "${task.submitAttempt==1 : 'TowerForge-queue-id-spot' : 'TowerForge-queue-id-on-demand'}"
+  script:
+  '''
+  your_job --here
+  '''
+}
+```
+
+<br>
+
+##### Setting the compute environment on pipeline creation
+When you are creating a new pipeline for your project through the web UI you have an
+option to set the Compute environment. Consider setting it to the `spot`
+environment if you are able.
+
+![compute-environment](./compute-environment.png)
+
+<br>
+
+### Optimize Memory and CPU Allocation
+
+Properly allocating memory and CPU resources is essential for cost optimization. Over-provisioning resources can lead to unnecessary expenses, while under-provisioning can result in slower performance and potential failures.
+
+**Best Practice:** Review the documentation for your specific workflow to determine the initial recommendations for memory and CPU allocation. Adjust these settings based on the workflow's requirements and monitor the performance to make further adjustments if necessary.
+
+#### Utilize the "optimize-nextflow" Project
+
+After running your workflow, leverage the "optimize-nextflow" project to analyze your runs. This tool provides suggestions for the appropriate CPU and memory settings for future runs, helping you to fine-tune your resource allocation.
+
+**Best Practice:** Regularly use "optimize-nextflow" to review your workflow runs. Implement the suggested optimizations to ensure you are using resources efficiently, which can lead to significant cost savings over time.
+
+### How-to adjust memory on memory related failures
+If you have nextflow processes that have wildly different memory requirements you may
+kick off a pipeline that is destined to fail. Use the following within your
+`.config` file to automatically retry the task and adjust the memory on each failure:
+
+```
+process {
```

Comment:
todo: Add a section that details that we can use labels here:
https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/blob/6b62e1b57e79273dfc96f16fcf7eabe4a6b4f4f3/nextflow.config#L57-L60